cloud/services/scalesets/vmss.go

```diff
@@ -177,55 +152,58 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 		},
 	}
 
-	_, err = s.Client.Get(ctx, vmssSpec.ResourceGroup, vmssSpec.Name)
-	if !azure.ResourceNotFound(err) {
-		if err != nil {
-			return errors.Wrapf(err, "failed to get scale set %s in %s", vmssSpec.Name, vmssSpec.ResourceGroup)
-		}
-		// scale set already exists, update it
+	existingVMSS, err := s.getExisting(ctx, vmssSpec.Name)
+	switch {
+	case err != nil && !azure.ResourceNotFound(err):
+		return errors.Wrapf(err, "failed to get VMSS %s", vmssSpec.Name)
+	case err == nil:
+		// VMSS already exists
+		// update the status.
+		s.Scope.SetProviderID(fmt.Sprintf("azure:///%s", existingVMSS.ID))
+		s.Scope.SetAnnotation("cluster-api-provider-azure", "true")
```

Comment:
should this be one of the well-defined labels? or what's going on here? I see it's the same before the diff, just nosing around

Reply:
confirmed there are actually not used anymore https://kubernetes.slack.com/archives/C8TSNPY4T/p1597858577173900

I'll open a follow-up PR to remove them (would rather do it separately from this huge refactor PR in case it needs to be reverted) from Machine + MachinePool